rusk-wallet/src/lib/cache.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use dusk_bytes::{DeserializableSlice, Serializable};
+use phoenix_core::Note;
+use rusqlite::{params, Connection};
+
+pub(crate) fn open(data_dir: &PathBuf) -> Connection {
+    let db_path = Path::new(data_dir).join("cache.db");
+
+    let cache = Connection::open(db_path).unwrap();
+    //create table if not exists TableName (col1 typ1, ..., colN typN)
+    cache
+        .execute(
+            "CREATE TABLE if not exists notes (note BLOB, spendkey BLOB)",
+            [],
+        )
+        .unwrap();
+    cache
+        .execute(
+            "CREATE TABLE if not exists cache (block BIGINT, spendkey BLOB)",
+            [],
+        )
+        .unwrap();
+    cache
+}
+
+pub(crate) fn last_blockheight(conn: &Connection, spendkey: &[u8]) -> u64 {
+    conn.query_row_and_then(
+        "SELECT block FROM cache WHERE spendkey = ?",
+        [spendkey],
+        |row| row.get(0),
+    )
+    .unwrap_or(0u64)
+}
+
+pub(crate) fn persist_blockheight(
+    conn: &Connection,
+    spendkey: &[u8],
+    actual_block: u64,
+) {
+    if conn
+        .execute(
+            "update cache set block=? WHERE spendkey = ?",
+            params!(actual_block, spendkey),
+        )
+        .unwrap()
+        == 0
+    {
+        conn.execute(
+            "INSERT INTO cache (block, spendkey) VALUES (?1, ?2)",
+            params!(actual_block, spendkey),
+        )
+        .unwrap();
+    }
```

Comment:
Maybe the only thing we can do is to notify the user with a warning and avoid any commit for this round (to avoid inconsistency)

Going to implement a transactional commit scheme for the whole "getBalance".